### PR TITLE
[IMP] pos*: multiple self and preparation display improvements

### PR DESCRIPTION
**Pull request by adgu-odoo**
pos*: point_of_sale, pos_self_order, pos_online_payment_self_order
This PR contains multiple improvement:

---
**1 pos_self_order: add quantity selector to product_info_popup**
In the self, when clicking on the information tag on a product, the
product_info_popup is displayed. In this popup, the user has
the possibility to add the product to its cart. However, the
quantity is always set to 1. This PR adds a quantity selector
to the product_info_popup so that the user can choose the quantity
he wants to add to its cart. For the products that have attributes or
are combos, the quantity selector is not displayed as it will be
displayed later in the process after clicking on add to cart.

---
**2 pos_online_payment_self_order, pos_self_order: change order_widget**
We change the way the order_widget acts. Now, if we are
in mobile with pay after meal, the order_widget will show 'To Order'
instead of 'To Pay' when clicking on 'Order Now' for a second time
and the order button will be disable until a new line is added. To see
what is already sent to the waiters, we can click on the 'My Order'
button in the landing page which will display the cart.

---
**3 pos_self_order: remove resume section for attribute_selection**
We remove the resume section for the attribute_selection
as it can lead to a lot of screen for the end-user (especially if the
attribute_selection is used in a combo). We add the add to cart button
instead with a quantity selector. The end resume is kept for the combo
selection.

---
**4 point_of_sale: add tracking_number as searchable field in ticket_screen**
We add the tracking_number of an order as a searchable
field in the ticket_screen. This is useful when using pos_self_order
for example. If a mobile is made with service at table and pay at
cashier after each order, the cashier can search the order by its
tracking_number in the history of orders.

---
https://github.com/odoo/enterprise/pull/49200
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_self_order/static/src/app/components/product_info_popup/product_info_popup.xml [hunk 1]
```diff
@@ -10,8 +10,17 @@
...
+                        <div class="modal-footer d-flex flex-row-reverse justify-content-between align-items-center">
                             <button type="button" class="btn btn-primary" t-on-click.stop="() => this.addToCartAndClose()">Add to Cart</button>
+                            <div t-if="!props.product.isCombo and !props.product.attributes.length > 0" class="o_self_order_incr_button btn-group " role="group" aria-label="Quantity select" >
+                                <button type="button"
+                                    t-on-click = "() => this.changeQuantity(false)"
+                                    class="btn btn-secondary"><span class="fs-2 lh-1 fa-fw d-inline-block">－</span></button>
+                                <div class="o-so-tabular-nums d-flex align-items-center px-3 text-bg-200" t-esc="state.qty"/>
+                                <button type="button"
+                                    t-on-click = "() => this.changeQuantity(true)"
+                                    class="btn btn-secondary"><span class="fs-2 lh-1 fa-fw d-inline-block">＋</span></button>
```
Comment (vlst-odoo) [inline addons/pos_self_order/static/src/app/components/product_info_popup/product_info_popup.xml:22]: could you use [this](https://github.com/odoo/odoo/blob/master/addons/point_of_sale/static/src/app/generic_components/inputs/numeric_input/numeric_input.js#L5) component here?

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/models/pos_order.py [hunk 2]
```diff
@@ -328,7 +328,17 @@ def _get_pos_anglo_saxon_price_unit(self, product, partner_id, quantity):
...
-    tracking_number = fields.Char(string="Short order number", compute='_compute_tracking_number')
+    tracking_number = fields.Char(string="Order Number", compute='_compute_tracking_number', search='_search_tracking_number')
+
+    def _search_tracking_number(self, operator, value):
+        #search is made over the pos_reference field
+        #The pos_reference field is like 'Order 00001-001-0001'
+        if operator == 'ilike':
+            if value[0] == '%' and value[-1] == '%':
+                value = value[1:-1]
+            value = value.zfill(3)
```
Comment (vlst-odoo) [inline addons/point_of_sale/models/pos_order.py:339]:
```suggestion
        if operator != "=" or not isinstance(value,str) or len(value) != 3:
            return
```

File: addons/pos_self_order/static/src/app/components/product_info_popup/product_info_popup.xml (hunk 1 above)
Comment (adgu-odoo) [reply addons/pos_self_order/static/src/app/components/product_info_popup/product_info_popup.xml:22]: This is a good idea, I'll try to do it after the freeze

**Comment by vlst-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**